tools/clang/unittests/HLSL/PixTestUtils.cpp

```diff
@@ -274,6 +274,66 @@ PassOutput RunAnnotationPasses(dxc::DxcDllSupport &dllSupport, IDxcBlob *dxil,
           Tokenize(outputText.c_str(), "\n")};
 }
 
+class InstructionOffsetSeekerImpl : public InstructionOffsetSeeker {
+
+  std::map<std::wstring, DWORD> m_labelToInstructionOffset;
+
+public:
+  InstructionOffsetSeekerImpl(DebuggerInterfaces &debuggerInterfaces) {
+    DWORD SourceFileOrdinal = 0;
+    CComBSTR fileName;
+    CComBSTR fileContent;
+    while (SUCCEEDED(debuggerInterfaces.compilationInfo->GetSourceFile(
+        SourceFileOrdinal, &fileName, &fileContent))) {
+      auto lines = strtok(std::wstring(fileContent), L"\n");
+      for (size_t line = 0; line < lines.size(); ++line) {
+        const wchar_t DebugLocLabel[] = L"debug-loc(";
+        auto DebugLocPos = lines[line].find(DebugLocLabel);
+        if (DebugLocPos != std::wstring::npos) {
+          auto StartLabelPos = DebugLocPos + (_countof(DebugLocLabel) - 1);
+          auto CloseDebugLocPos = lines[line].find(L")", StartLabelPos);
+          if (CloseDebugLocPos == std::string::npos) {
+            VERIFY_ARE_NOT_EQUAL(CloseDebugLocPos, std::string::npos);
+          }
+          auto Label = lines[line].substr(StartLabelPos,
+                                          CloseDebugLocPos - StartLabelPos);
+          CComPtr<IDxcPixDxilInstructionOffsets> InstructionOffsets;
+          if (FAILED((debuggerInterfaces.debugInfo
+                          ->InstructionOffsetsFromSourceLocation(
+                              fileName, line + 1, 0, &InstructionOffsets)))) {
+            VERIFY_FAIL(L"InstructionOffsetsFromSourceLocation failed");
+          }
+          auto InstructionOffsetCount = InstructionOffsets->GetCount();
+          if (InstructionOffsetCount == 0) {
+            VERIFY_FAIL(L"Instruction offset count was zero");
+          }
+          if (m_labelToInstructionOffset.find(Label) !=
+              m_labelToInstructionOffset.end()) {
+            VERIFY_FAIL(L"Duplicate label found!");
+          }
+          // Just the last offset is sufficient:
+          m_labelToInstructionOffset[Label] =
+              InstructionOffsets->GetOffsetByIndex(InstructionOffsetCount - 1);
+        }
+      }
+      SourceFileOrdinal++;
+      fileName.Empty();
+      fileContent.Empty();
```

Comment:
I think you'll need to add the `Empty()` method to the CComBSTR implementation in `WinAdapter.h` for this to work.  That implementation is super minimal, and doesn't seem to have a way to clear the string other than the destructor.
See here:
https://github.com/microsoft/DirectXShaderCompiler/blob/main/include/dxc/WinAdapter.h#L1000